### PR TITLE
#665 API to open and close inline-toolbar

### LINKED ISSUE #665: [Core API]: Provide methods that opens/closes inline toolbar

### PR DESCRIPTION
**Pull request by tanmayv**
This is regarding the issue #665. I tested this on `inline-tool-link` and `inlineToolbar` is available to close it.

Resolves #665

### CODE REVIEW

**Comment by neSpecc** [comment]
Resolves #665

File: types/api/index.d.ts [hunk 1]
```diff
@@ -8,3 +8,4 @@ export * from './styles';
 export * from './caret';
 export * from './toolbar';
 export * from './notifier';
+export * from './inline'
```
Comment (neSpecc) [inline types/api/index.d.ts]: It would better to name it `inlineToolbar`

File: src/components/modules/toolbar/inline.ts [hunk 2]
```diff
@@ -170,7 +170,7 @@ export default class InlineToolbar extends Module {
   /**
    * Shows Inline Toolbar
    */
-  private open(): void {
```
Comment (tanmayv) [inline src/components/modules/toolbar/inline.ts]: @neSpecc This file is also named `inline.ts`. Should we rename this as well?

Comment (neSpecc) [reply src/components/modules/toolbar/inline.ts]: no, it's included at the `/toolbar/` directory, so this is a redundant addition

**Comment by tanmayv** [comment]
Do i need to `dist/editor.js` as well? It was in `dist/*` so I skipped it.

**Comment by neSpecc** [comment]
> Do i need to `dist/editor.js` as well? It was in `dist/*` so I skipped it.

yes, currently we have a bundle in a repo. Maybe removed later.

**Comment by tanmayv** [comment]
Okay I'll push `dist/editor.js` as well.

**Comment by tanmayv** [comment]
@neSpecc Should I build bundle in `prod`, or `dev` is okay?

**Comment by gohabereg** [comment]
@tanmayv 
The example-dev.html page doesn't work in your branch

**Comment by tanmayv** [comment]
@gohabereg Fixed the issue. `src/components/modules/api/inline-toolbar.ts` can't have hyphen in it so change it to `src/components/modules/api/inlineToolbar.ts`

**Comment by neSpecc** [comment]
Also error throws event when one block is selected, but without caret in there. Click at the Block's right zone, near three-dots, then call  `open`

**Comment by tanmayv** [comment]
>     * open editor
> 
>     * do not select any block
> 
>     * call `api.inlineToolbar.open()`
> 
> 
> Error will be thrown
> 
> ```
> TypeError: null is not an object (evaluating 't.parentNode')
> value — editor.js:4566
> value — editor.js:6737
> value — editor.js:6712
> value — editor.js:3925
> Global Code
> evaluateWithScopeExtension
> _evaluateOn
> _evaluateAndWrap
> ```
> 
> You should add a checking for a `BlockManager.currentBlock` existence at the `toolbar/inline.js@open`

This should be fixed now.

**Comment by tanmayv** [comment]
> Also error are throwed even when one block is selected, but without caret in there. Click at the Block's right zone, near three-dots, then call `open`

I wasn't able to recreate this. Is this still happening?

**Comment by tanmayv** [comment]
Updated the `changelog.md` and `api.md` as well to document new api methods.

**Pull request merged**